src/lib/edf_schedule.c

```diff
@@ -468,6 +472,7 @@ static int edf_scheduler_init(void)
 	sch = sch_data->edf_sch_data;
 
 	list_init(&sch->list);
+	list_init(&sch->idle_list);
```

Comment:
Missing `list_item_del(&sch->idle_list)` in `edf_scheduler_free`.

Reply:
Good catch, thanks.